lib/model/model.go

```diff
@@ -2492,6 +2529,26 @@ func (m *model) CommitConfiguration(from, to config.Configuration) bool {
 	m.fmut.Unlock()
 	m.closeConns(removedDevices, errDeviceRemoved)
 
+	// Forget pending folder/device combinations that are now shared or ignored, plus
+	// any for our own device ID (should not happen, treat us like an unknown device)
+	delete(forgetPending, to.MyID)
```

Comment:
Will move to a new `UnmarkDevice()` method to hide the implementation detail.

> You could add an `if deviceID != m.id {` condition around the marking in the `toDevices` loop above to be save.

That's not enough unfortunately.  The very first loop over toFolders will definitely include our own device ID in the map, because `FolderConfiguration` always contains the local device.  Since we don't want any pending folders for our own device ID, it needs to be excluded completely, as we cannot produce an exhaustive drop list of folder IDs.  I messed around with filtering the `cfg.DeviceIDs()` output, but this one-liner approach seemed much more elegant and readable.

Note that because of the inverted semantics for `CleanPendingDevices()` regarding device IDs, MyID needs to be re-added before that second loop.

Reply:
Fixed (hopefully) in 232831a5d8abf7d180cc69bb314659b5403ca478.